syntax = "proto3";

package dataset;


message Empty {}

message XMLResponse {
  string xml_data = 1;
}

message XPathRequest {
  string expression = 1;
}

message XPathResponse {
  repeated string results = 1;
}

message InsertResponse {
  bool success = 1;
  string message = 2;
}


service DataService {

  // Devolve o XML completo
  rpc GetXML(Empty) returns (XMLResponse);

  // Executa uma express√£o XPath no XML
  rpc QueryXPath(XPathRequest) returns (XPathResponse);

  // Insere os dados do XML na base de dados PostgreSQL
  rpc InsertToDB(Empty) returns (InsertResponse);
}
